tests/lib/rules/no-sequences.js

```diff
@@ -65,6 +85,25 @@ ruleTester.run("no-sequences", rule, {
                 endColumn: 3
             }]
         },
+        { code: "var a = 1, b = 2, c = (d = 3, e = 4)", errors: errors(29) },
+        { code: "var a = b = 1, c = (d = 3, e = 4)", errors: errors(26) },
+        { code: "var a = (b = 1, c = 2)", errors: errors(15) },
+        { code: "var a = (b, c = 2)", errors: errors(11) },
+        { code: "var a = (b = 1, c)", errors: errors(15) },
+        { code: "let a = (b = 1, c = 2)", env: { es6: true }, errors: errors(15) },
+        { code: "let a = (b, c = 2)", env: { es6: true }, errors: errors(11) },
+        { code: "let a = (b = 1, c)", env: { es6: true }, errors: errors(15) },
+        { code: "const a = (b = 1, c = 2)", env: { es6: true }, errors: errors(17) },
+        { code: "const a = (b, c = 2)", env: { es6: true }, errors: errors(13) },
+        { code: "const a = (b = 1, c)", env: { es6: true }, errors: errors(17) },
+        { code: "const a = (b = c = 1, d)", env: { es6: true }, errors: errors(21) },
+        { code: "a = (b = 1, c = 2)", errors: errors(11) },
+        { code: "a = (b, c = 2)", errors: errors(7) },
+        { code: "a = (b = 1, c)", errors: errors(11) },
+        { code: "a = b = 1, c = 2", errors: errors(10) },
```

Comment:
Actually, I think we can make this check smarter by checking if it's a child of an `ExpressionStatement` or not. Will update.